bpf-gpl/tc.c

```diff
@@ -455,6 +455,13 @@ static CALI_BPF_INLINE int calico_tc(struct __sk_buff *skb)
 		}
 	}
 
+	// Drop packets from/to workload Endpoints with IP options
+	if (CALI_F_TO_WEP || CALI_F_FROM_WEP) {
```

Comment:
I think we should do this drop on all TC attach points. We discussed that we should not drop packet that are destined for the host itself, but this would allow the HEP program to drop a packet that is for a workload.

I think we would need to check, after we knew the ihl>5, whether it is for the host IP using routing lookup and `goto allow` as that would skip any further parsing of the packet. Unfortunately, that would give a blind cheque to any packet that is for the host and carries IP option and thus it might be used to bypass any policy in the future, but since we do not have any HEP policy yet, it does not matter much and we will address it once we have the policies and once we decide to handle options.

Reply:
This comment needs to be addresed

Reply:
I am not sure if we will know whether a packet is for the host IP before NAT. @fasaxc @tomastigera please confirm.